packages/plugin-betswirl/src/actions/coinToss.ts

```diff
@@ -0,0 +1,288 @@
+import { z } from "zod";
+import { type Hex } from "viem";
+import {
+    type Action,
+    type IAgentRuntime,
+    type Memory,
+    type HandlerCallback,
+    type State,
+    elizaLogger,
+    type ActionExample,
+    composeContext,
+    generateObject,
+    ModelClass,
+} from "@moxie-protocol/core";
+import { MoxieWalletClient } from "@moxie-protocol/moxie-agent-lib/src/wallet";
+import {
+    CASINO_GAME_TYPE,
+    CoinToss,
+    COINTOSS_FACE,
+    slugById,
+    formatTxnUrl,
+} from "@betswirl/sdk-core";
+import { casinoBetParams, getMaxBetCountParam } from "../types";
+import {
+    getChainIdFromWallet,
+    getBetToken,
+    placeBet,
+    getBet,
+    getBetAmountInWei,
+} from "../utils/betswirl";
+import { formatTokenForMoxieTerminal } from "../utils/moxie";
+
+export const CoinTossBetParameters = z.object({
+    face: z
+        .nativeEnum(COINTOSS_FACE)
+        .nullable()
+        .describe("The face of the coin"),
+    ...casinoBetParams,
+    ...getMaxBetCountParam(CASINO_GAME_TYPE.COINTOSS),
+});
+export const coinTossTemplate = `
+Extract the following details to flip a coin:
+- **betAmount** (String?): The amount to wager.
+- **face** (String?): The side of the coin to bet on. Can be either:
+  - HEADS
+  - TAILS
+- **token** (String?): The token symbol.
+Where "?" indicates that the value is optional.
+
+Provide the values in the following JSON format:
+\`\`\`json
+{
+    "betAmount": string?,
+    "face": string?,
+    "token": string?
+}
+\`\`\`
+
+Here are example messages and their corresponding responses:
+
+**Message 1**
+\`\`\`
+Bet 0.01 ETH on heads
+\`\`\`
+
+**Response 1**
+\`\`\`json
+{
+    "betAmount": "0.01",
+    "face": "heads",
+    "token" "ETH"
+}
+\`\`\`
+
+**Message 2**
+\`\`\`
+Double or nothing 0.5 ETH on heads
+\`\`\`
+
+**Response 2**
+\`\`\`json
+{
+    "betAmount": "0.5",
+    "face": "HEADS",
+    "token": "ETH",
+}
+\`\`\`
+
+** Message 3 **
+\`\`\`
+Flip a coin for me
+\`\`\`
+
+** Response 3 **
+\`\`\`json
+{
+    "betAmount": null,
+    "face": null,
+    "token": null,
+}
+\`\`\`
+
+** Message 4 **
+\`\`\`
+I want to bet on the tails side
+\`\`\`
+
+** Response 4 **
+\`\`\`json
+{
+    "betAmount": null,
+    "face": "TAILS",
+    "token": null,
+}
+\`\`\`
+
+** Message 5 **
+\`\`\`
+I want to bet 0.01 on heads
+\`\`\`
+
+** Response 5 **
+\`\`\`json
+{
+    "betAmount": "0.01",
+    "face": "HEADS",
+    "token": null,
+}
+\`\`\`
+
+** Message 6 **
+\`\`\`
+I want to bet my ETH on tails
+\`\`\`
+
+** Response 6 **
+\`\`\`json
+{
+    "betAmount": null,
+    "face": "TAILS",
+    "token": "ETH",
+}
+\`\`\`json
+
+Here are the recent user messages for context:
+{{recentMessages}}
+`;
+export const coinTossAction: Action = {
+    name: "COIN_TOSS",
+    similes: [
+        "COIN_FLIP",
+        "DOUBLE_OR_NOTHING",
+        "TOSS_A_COIN",
+        "BETSWIRL_COIN_TOSS",
```

Comment:
In addition to adding examples below, also add more `similes` to help the training for the agent:

```suggestion
        "COIN_FLIP",
        "DOUBLE_OR_NOTHING",
        "TOSS_A_COIN",
        "BETSWIRL_COIN_TOSS",
        "BET_ON_HEADS",
        "BET_ON_TAILS",
        "BET_ON_COIN_FLIP",
        "BET_ON_COIN_TOSS",
        "BET_ON_HEADS_BETSWIRL",
        "BET_ON_TAILS_BETSWIRL",
        "BET_ON_CON_FLIP_BETSWIRL",
        "BET_ON_COIN_TOSS_BETSWIRL"
```